tests/components/saunum/test_climate.py

```diff
@@ -0,0 +1,412 @@
+"""Test the Saunum climate platform."""
+
+from __future__ import annotations
+
+from unittest.mock import AsyncMock, patch
+
+from pysaunum import SaunumData
+import pytest
+
+from homeassistant.components.climate import (
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_HVAC_ACTION,
+    ATTR_HVAC_MODE,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    DOMAIN as CLIMATE_DOMAIN,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_TEMPERATURE,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.const import ATTR_ENTITY_ID, ATTR_TEMPERATURE, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def platforms() -> list[Platform]:
+    """Fixture to specify platforms to test."""
+    return [Platform.CLIMATE]
+
+
+@pytest.mark.usefixtures("init_integration")
+async def test_climate_entity_creation(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test climate entity creation."""
+    # Check climate entity is created
+    entity_id = "climate.saunum_leil"
+    entity_entry = entity_registry.async_get(entity_id)
+    assert entity_entry is not None
+
+    state = hass.states.get(entity_id)
+    assert state is not None
```

Comment:
Checkout snapshot tests. If you use `snapshot_platform` you can run `pytest ./tests/components/saunum --snapshot-update` and it will generate .ambr files with the whole thing fixated in a file